modules/extensions/galasa-extensions-parent/dev.galasa.ras.couchdb/src/main/java/dev/galasa/ras/couchdb/internal/CouchdbDirectoryService.java

```diff
@@ -323,6 +324,39 @@ private CouchdbRunResult fetchRun(String id) throws ParseException, IOException,
         return tests;
     }
 
+    @Override
+    public List<IRunResult> getRunsByGroupName(@NotNull String groupName) throws ResultArchiveStoreException {
+
+    
+        List<IRunResult> runs = new ArrayList<>();
+        try {
+            boolean includeDocuments = true;
+            ViewResponse viewResponse = store.getDocumentsFromDatabaseViewByKey(
+                RUNS_DB,
+                RUN_GROUP_VIEW_NAME,
+                groupName,
+                includeDocuments
+            );
+
+            if (viewResponse.rows == null) {
+                String errorMessage = ERROR_FAILED_TO_GET_VIEW_DOCUMENTS_FROM_DATABASE.getMessage(RUN_GROUP_VIEW_NAME, RUNS_DB);
+                throw new ResultArchiveStoreException(errorMessage);
+            }
+
+            for (ViewRow row : viewResponse.rows) {
+                if (row.doc != null) {
+                    JsonObject testStructureJson = gson.toJsonTree(row.doc).getAsJsonObject();
+                    TestStructureCouchdb testStructure = gson.fromJson(testStructureJson, TestStructureCouchdb.class);
+                    runs.add(new CouchdbRunResult(store, testStructure, logFactory));
+                }
+            }
+        } catch (CouchdbException e) {
+            // This error has a custom message, so pass it up
+            throw new ResultArchiveStoreException(e);
+        }
+        return runs;
```

Comment:
This is duplicated from the `getRunsByRunName` code, the only differences being the view name and the group being passed in instead of the run name - might be worth making a function that gets runs from a view and use that function in both `getRunsByRunName` and `getRunsByGroupName`